device/core/src/blob_upload/blob_upload_result.ts

```diff
@@ -66,4 +67,24 @@ export class BlobUploadResult {
 
     return uploadResult;
   }
+
+  static fromAzureStorageCallbackArgs2(uploadResponse: BlobUploadCommonResponseStub): BlobUploadResult {
```

Comment:
>fromAzureStorageCallbackArgs2 [](start = 9, length = 29)

versioning APIs in names is bad practice. you can probably see for yourself a lot of duplicate code between this API and the original one. 
You should strive to minimize the amount of code you write because you'll have to maintain it.

the initial API isn't public. is there a way to modify it so that it works in both cases?